pkg/controllers/repositoryserver/handler.go

```diff
@@ -171,20 +168,26 @@ func (h *RepoServerHandler) reconcilePod(ctx context.Context, svc *corev1.Servic
 		return nil, nil, err
 	}
 	h.Logger.Info("Pod resource not found. Creating new pod")
+	return h.createPodUpdateStatus(ctx, repoServerNamespace, svc)
+}
+
+func (h *RepoServerHandler) setCondition(ctx context.Context, reason, conditionMsg, conditionType string, progress crv1alpha1.RepositoryServerProgress, status metav1.ConditionStatus) error {
```

Comment:
I think we should accept these parameters as a struct type, maybe with name `Condition`.

Reply:
the function set the condition and also updates the status, do you mean creating condition struct including `progress` as well?

Reply:
The function has 6 parameters now. My recommendation was to create a struct with these parameters so that the function would accept that type as parameter

```
type Condition struct{
reason, 
conditionMsg,
 conditionType string, 
progress crv1alpha1.RepositoryServerProgress, 
status metav1.ConditionStatus
}

func (h *RepoServerHandler) setCondition(ctx context.Context, c Condition){
}
```

Reply:
updated